lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+					source,
+					groupAssociations,
+				] of allAssociations.entries()) {
+					for (const [groupId, associations] of groupAssociations) {
+						if (associations.length > 0) {
+							await this.driver.controller.removeAssociations(
+								source,
+								groupId,
+								associations as AssociationAddress[]
+							)
+							logger.info(
+								`Assocaitions: Removed ${
+									associations.length
+								} associations from Node ${
+									source.nodeId +
+									(source.endpoint
+										? ' Endpoint ' + source.endpoint
+										: '')
+								} group ${groupId}`
+							)
+						}
+					}
+				}
+			} catch (error) {
+				logger.warn(
+					`Error while removing all associations from ${nodeId}: ${error.message}`
+				)
+			}
+		} else {
+			logger.warn(
+				`Node ${nodeId} not found when calling 'removeAllAssociations'`
+			)
+		}
+	}
+
+	/**
+	 * Remove node from all associations
+	 *
+	 * @param {number} nodeId Zwave node id
+	 */
+	async removeNodeFromAllAssociations(nodeId: number) {
+		const zwaveNode = this.getNode(nodeId)
+
+		if (zwaveNode) {
+			try {
+				logger.info(
+					`Assocaitions: Removing Node ${nodeId} from all associations`
+				)
+				await this.driver.controller.removeNodeFromAllAssociations(
+					nodeId
+				)
+			} catch (error) {
+				logger.warn(
+					`Error while removing Node ${nodeId} from all associations: ${error.message}`
+				)
+			}
+		} else {
+			logger.warn(
+				`Node ${nodeId} not found when calling 'removeNodeFromAllAssociations'`
+			)
+		}
+	}
+
+	/**
+	 * Refresh all nodes neighbors
+	 */
+	async refreshNeighbors(): Promise<Record<number, number[]>> {
+		if (this._lockNeighborsRefresh === true) {
+			throw Error('you can refresh neighbors only once every 60 seconds')
+		}
+
+		this._lockNeighborsRefresh = true
+
+		// set the timeout here so if something fails later we don't keep the lock
+		setTimeout(
+			() => (this._lockNeighborsRefresh = false),
+			NEIGHBORS_LOCK_REFRESH
+		)
+
+		const toReturn = {}
+		// when accessing the controller memory, the Z-Wave radio must be turned off with to avoid resource conflicts and inconsistent data
+		await this.driver.controller.toggleRF(false)
+		for (const [nodeId, node] of this.nodes) {
+			try {
+				node.neighbors = (await this.getNodeNeighbors(
+					nodeId,
+					true
+				)) as number[]
+			} catch (error) {
+				logger.error(error)
+			}
+			toReturn[nodeId] = node.neighbors
+		}
+		// turn rf back to on
+		await this.driver.controller.toggleRF(true)
+
+		return toReturn
+	}
+
+	/**
+	 * Get neighbors of a specific node
+	 */
+	async getNodeNeighbors(
+		nodeId: number,
+		dontThrow: boolean
+	): Promise<Readonly<number[]>> {
+		let neighbors: Readonly<number[]>
+		try {
+			neighbors = await this.driver.controller.getNodeNeighbors(nodeId)
+		} catch (error) {
+			logger.error(
+				`Node ${nodeId} error while updating Neighbors: ${error.message}`
+			)
+			if (!dontThrow) {
+				throw error
+			}
+		}
+
+		return neighbors
```

Comment:
Much simpler and without the duplicate type annotation:
```suggestion
		try {
			return await this.driver.controller.getNodeNeighbors(nodeId)
		} catch (error) {
			logger.error(
				`Node ${nodeId} error while updating Neighbors: ${error.message}`
			)
			if (!dontThrow) {
				throw error
			}
		}
```

Reply:
Your implementation can return `undefined` though (if an error happens and `dontThrow` is true). The return type does not reflect that.